infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -414,30 +417,38 @@ private void registerToLogReplicationLock() {
      * as source (sender/producer) or sink (receiver).
      */
     private void onLeadershipAcquire() {
-        switch (topologyDescriptor.getLocalClusterDescriptor().getRole()) {
-            case SOURCE:
-                log.info("Start as Source (sender/replicator)");
-                sessionManager.startReplication();
-                lockAcquireSample = recordLockAcquire(topologyDescriptor.getLocalClusterDescriptor().getRole());
-                processCountOnLockAcquire();
-                break;
-            case SINK:
-                log.info("Start as Sink (receiver)");
-
-                // Sink Site : the LogReplicationServer (server handler) will
-                // reset the LogReplicationSinkManager on acquiring leadership
-                interClusterServerNode.setLeadership(true);
-                lockAcquireSample = recordLockAcquire(topologyDescriptor.getLocalClusterDescriptor().getRole());
-
-                processCountOnLockAcquire();
-                break;
-            default:
-                log.error("Log Replication not started on this cluster. Leader node {} id {} belongs to cluster with {} role.",
-                        localEndpoint, localNodeId, topologyDescriptor.getLocalClusterDescriptor().getRole());
-                break;
+        if (isSource()) {
+            log.info("Start as Source (sender/replicator)");
+            // TODO: Shama: currently the supported model is FULL_TABLE. So fetchConnectionEndpoints will get all SINKs.
+            //  This will change in the connection Model PR where an endpoint can be SOURCE/SINK.
+            sessionManager.startReplication(fetchConnectionEndpoints());
+            lockAcquireSample = recordLockAcquire(topologyDescriptor.getLocalClusterDescriptor().getRole());
+            processCountOnLockAcquire();
+        } else if (isSink()) {
```

Comment:
same here, on leadership acquire it can be both right? source and sink for different models. Or by the comment it meas this comes as part of the other PR?

Reply:
Yeah, the comment means that it will come in the connectionModel PR